jetty-ee9/jetty-ee9-nested/src/main/java/org/eclipse/jetty/ee9/nested/ResourceHandler.java

```diff
@@ -97,12 +105,33 @@ public void doStart() throws Exception
         if (_mimeTypes == null)
             _mimeTypes = _context == null ? new MimeTypes() : _context.getMimeTypes();
 
-        _resourceService.setContentFactory(new ResourceContentFactory(this, _mimeTypes, _resourceService.getPrecompressedFormats()));
+        _byteBufferPool = getByteBufferPool(_context);
+        _resourceService.setContentFactory(setupContentFactory());
         _resourceService.setWelcomeFactory(this);
 
         super.doStart();
     }
 
+    private static ByteBufferPool getByteBufferPool(ContextHandler contextHandler)
+    {
+        if (contextHandler == null)
+            return new NoopByteBufferPool();
+        Server server = contextHandler.getServer();
+        if (server == null)
+            return new NoopByteBufferPool();
+        ByteBufferPool byteBufferPool = server.getBean(ByteBufferPool.class);
+        return (byteBufferPool == null) ? new NoopByteBufferPool() : byteBufferPool;
+    }
+
+    protected HttpContent.Factory setupContentFactory()
+    {
+        HttpContent.Factory contentFactory = new ResourceHttpContentFactory(this, _mimeTypes);
+        contentFactory = new PreCompressedHttpContentFactory(contentFactory, _resourceService.getPrecompressedFormats());
+        contentFactory = new FileMappedHttpContentFactory(contentFactory);
+        contentFactory = new ValidatingCachingContentFactory(contentFactory, Duration.ofSeconds(1).toMillis(), _byteBufferPool);
+        return contentFactory;
+    }
```

Comment:
Please also add `get/setHttpContentFactory()` so that it's easier to configure before `start()` (does not require subclassing).

Reply:
I have now created these methods, and `setupHttpContentFactory()` is only called if there was no `HttpContentFactory` set manually.